knative-operator/pkg/controller/knativeserving/consoleclidownload/consoleclidownload.go

```diff
@@ -44,52 +41,57 @@ func Apply(instance *servingv1alpha1.KnativeServing, apiclient client.Client, sc
 	return nil
 }
 
-// reconcileKnCCDResources reconciles required resources viz Deployment, Service, Route
+// reconcileKnCCDResources reconciles required resources viz Knative Service
 // which will serve kn cross platform binaries within cluster
 func reconcileKnCCDResources(instance *servingv1alpha1.KnativeServing, apiclient client.Client, scheme *runtime.Scheme) error {
 	log.Info("Installing kn ConsoleCLIDownload resources")
-	manifest, err := manifest(instance, apiclient, scheme)
-	if err != nil {
+	serviceFromCluster := &servingv1.Service{}
+	service := populateKnService(os.Getenv("IMAGE_KN_CLI_ARTIFACTS"), instance)
+	err := apiclient.Get(context.TODO(), client.ObjectKey{Namespace: instance.GetNamespace(), Name: knConsoleCLIDownloadService}, serviceFromCluster)
+	switch {
+	case apierrors.IsNotFound(err):
+		if err := apiclient.Create(context.TODO(), service); err != nil {
+			return err
+		}
+	case err == nil:
+		serviceFromClusterDC := serviceFromCluster.DeepCopy()
+		changed := false
+		if serviceFromCluster.Spec.Template.Spec.GetContainer().Image != service.Spec.Template.Spec.GetContainer().Image {
+			serviceFromClusterDC.Spec.Template.Spec.GetContainer().Image = service.Spec.Template.Spec.GetContainer().Image
+			changed = true
+		}
+		if !equality.Semantic.DeepEqual(serviceFromCluster.Spec.Template.Spec.GetContainer().Resources, service.Spec.Template.Spec.GetContainer().Resources) {
+			serviceFromClusterDC.Spec.Template.Spec.GetContainer().Resources = service.Spec.Template.Spec.GetContainer().Resources
+			changed = true
+		}
```

Comment:
Why are we checking resources here?

Reply:
My thought was to check if fields needs to be updated when `Service` exists already.

Reply:
Ah, one would usually just check `Spec` for equality, but I guess this works too :shrug:

Reply:
Sounds more reasonable and streamlined, thanks for the suggestion.